systest/tests/transactions_test.go

```diff
@@ -0,0 +1,92 @@
+package tests
+
+import (
+	"testing"
+
+	spacemeshv1 "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/systest/cluster"
+	"github.com/spacemeshos/go-spacemesh/systest/testcontext"
+)
+
+func testTransactions(t *testing.T, tctx *testcontext.Context, cl *cluster.Cluster) {
+	const (
+		keys        = 10
+		genesis     = 7
+		sendFor     = 8
+		stopSending = genesis + sendFor + 1
+		stopWaiting = stopSending + 2
+		batch       = 1
+		amount      = 100
+	)
+	receiver := [20]byte{11, 1, 1}
+
+	eg, ctx := errgroup.WithContext(tctx)
+	for i := 0; i < keys; i++ {
+		client := cl.Client(i % cl.Total())
+		submitter := newTransactionSubmitter(cl.Private(i), receiver, amount, client)
+		collectLayers(ctx, eg, client, func(layer *spacemeshv1.LayerStreamResponse) (bool, error) {
```

Comment:
maybe don't call this collectLayers, as you are submitting TXs.

Reply:
right, the idea is to have a generic function that watches layers. probably `watchLayers` will be a better name